### PR TITLE
Separate the refund import process in multiple objects

### PR DESCRIPTION
**Pull request by thisiscab**
I've been toying around, trying to figure out a better way to structure
the code surrounding the Shopify refund importer.

The code is not namespaced correctly but this should give a general idea
on how I was thinking about stuff.

### CODE REVIEW

File: app/importers/spree/retail/shopify/return_items.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module Spree
+  module Retail
+    module Shopify
+      class ReturnItems
```
Comment (thisiscab) [inline app/importers/spree/retail/shopify/return_items.rb]: I don't like this class! I feel like it require too much knowledge for what it does.

File: refunder_logger.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Retail
+    module Shopify
+      class RefunderLogger
```
Comment (thisiscab) [inline refunder_logger.rb]: I would abstract stuff into a General Logger that specific logger would inherit of. 
That way the specific logger would be more explicit in what it does.

File: app/controllers/spree/retail/shopify/hooks/refunds_controller.rb [hunk 3]
```diff
@@ -5,9 +5,15 @@ module Hooks
         class RefundsController < HooksController
           def create
             shopify_refund = ShopifyAPI::Refund.find(shopify_refund_id, params: { order_id: shopify_order_id })
-            success = Spree::Retail::Shopify::GenerateRefundOrder.new(shopify_refund).process
+            Spree::Retail::Shopify::RefundImporter.new(shopify_refund, self).perform
+          end
+
+          def success_case
```
Comment (thisiscab) [inline app/controllers/spree/retail/shopify/hooks/refunds_controller.rb:13]: I remember seeing something like that in Niko's code, I found that interesting. I'm interested in knowing how it pans out!

**Comment by thisiscab** [comment]
Hey! If you guys are down to give me comments or ideas or anything, that would be great.

It's not yet functional but it should give a good idea of what I'm aiming for. Also those classes are so small that it should easier to tests in contrary of what's currently there.

**Comment by thisiscab** [comment]
@mickeyren Hey, I'd like to get this merged into master, would you mind doing a quick CR please?

**Comment by papayaah** [comment]
@cabouffard  nothing sinister looking out of the blue - but then again not sure if i'm the right person to look into reviewing this. how about @bryanmtl eyes?

**Comment by thisiscab** [comment]
In my mind, if the specs still passes, it means that it will still work has expected. I think we are good to go 🗡

**Comment by thisiscab** [comment]
@bryanmtl May want to hold off though, since he's currently testing.

**Pull request merged**